controllers/managedocs_controller_test.go

```diff
@@ -46,14 +45,67 @@ var _ = Describe("ManagedOCS controller", func() {
 	}
 
 	Context("reconcile()", func() {
-		When("there is no storage cluster resource in the cluster", func() {
-			It("should create a new storage cluster", func() {
+		When("There is no addonParam secret in the cluster", func() {
+			It("should not create a new storage cluster", func() {
+				ctx := context.Background()
+
+				scList := &ocsv1.StorageClusterList{}
+
+				Expect(k8sClient.List(ctx, scList, client.InNamespace(TestNamespace))).Should(Succeed())
+				Expect(scList.Items).Should(HaveLen(0))
+
+				// addon param secret does not exist
+				secret := &corev1.Secret{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      AddOnParamsSecretName,
+						Namespace: TestNamespace,
+					},
+				}
+				Expect(k8sClient.Get(ctx, getResourceKey(secret), secret)).Should(
+					WithTransform(errors.IsNotFound, BeTrue()))
+
+				managedOCS := &v1.ManagedOCS{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      ManagedOCSName,
+						Namespace: TestNamespace,
+					},
+				}
+				Expect(k8sClient.Create(ctx, managedOCS)).Should(Succeed())
+				Expect(k8sClient.Get(ctx, getResourceKey(managedOCS), managedOCS)).Should(Succeed())
+
+				defer func() {
+					Expect(k8sClient.Delete(context.Background(), managedOCS)).Should(Succeed())
+				}()
+				// No storage cluster should be created
+				scList = &ocsv1.StorageClusterList{}
+
+				Expect(k8sClient.List(ctx, scList, client.InNamespace(TestNamespace))).Should(Succeed())
+				Expect(scList.Items).Should(HaveLen(0))
+
+			})
+		})
+		When("there is incorrect data in the secret", func() {
+			It("should not create a new storage cluster", func() {
```

Comment:
This should never be a case in regards to size unless you are treating a string that does no contain a number a case.

Reply:
Modified with the change in expected secret input